app/models/host/managed.rb

```diff
@@ -129,7 +129,6 @@ class Jail < ::Safemode::Jail
   alias_attribute :os, :operatingsystem
   alias_attribute :arch, :architecture
 
-  validates :environment_id,  :presence => true
```

Comment:
Maybe like this to avoid the case you mentioned?

`validates :environment_id, :presence => true, :unless => Proc.new { |host| host.puppet_proxy_id.blank? }`